openai-java-core/src/main/kotlin/com/openai/models/chat/completions/ChatCompletionMessageToolCall.kt

```diff
@@ -258,6 +259,18 @@ private constructor(
          */
         fun arguments(): String = arguments.getRequired("arguments")
 
+        /**
+         * Gets the arguments to the function call, converting the values from the model in JSON
+         * format to an instance of a class that holds those values. The class must previously have
+         * been used to define the JSON schema for the function definition's parameters, so that the
+         * JSON corresponds to structure of the given class.
+         *
+         * @see ChatCompletionCreateParams.Builder.addTool
+         * @see arguments
+         */
+        fun <T> arguments(functionParametersType: Class<T>): T =
```

Comment:
For this and in the other file, might be nice to note `@throws OpenAIInvalidDataException` with an explanation of when that happens (i.e. API bug or you passed a different class vs what you told the server)